src/components/Details/Details.js

```diff
@@ -1,43 +1,61 @@
 import React from 'react'
 import './Details.css'
 
-const Details = ({selectedMovie, returnHome}) => {
-    
-    const {
-        id,
-        backdrop_path,
-        overview,
-        release_date,
-        average_rating,
-        genres,
-        budget,
-        revenue,
-        runtime,
-        tagline,
-        title,
-        } = selectedMovie;
+class Details extends React.Component {
+    constructor(){
+        super()
 
-    return (
-        <div 
-            className='details-card'
-            id={id}
-        >
-            <button onClick={() => returnHome()}>Go Back</button>
-            <h2>{title}</h2>
-            <p>{tagline}</p>
-            <img src={backdrop_path} alt={`${title} backdrop`}/>
-            <h3>Description</h3>
-            <p>{overview}</p>
-            <p>Release Date: {release_date}</p>     
-            <p>{average_rating.toFixed(1)} ⭐️</p>
-            <p>Film genre: {genres}</p>
-            <p>Film budget: ${budget}</p>
-            <p>Box Office Revenue : ${revenue}</p> {// Round to millions 
-            }
-            <p>{runtime} minutes</p>
-            
-        </div>
-    )
-}
+        this.state = {
+            errorMessage: '',
+            movie: {},
+            movieTrailerUrl: {},
+        };
+    }
 
+    componentDidMount() {
+        fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${this.props.selectedMovie.id}`)
+        .then(response => response.json())
+        .then(data => {
+            let movie = data.movie
+            this.setState({movie})
+          })
+          .catch(err => {
```

Comment:
Since we're using this in App too, would it make sense to create a separate function for displaying errors?